solc/CommandLineInterface.cpp

```diff
@@ -667,43 +692,44 @@ bool CommandLineInterface::compile()
 	{
 		if (m_options.metadata.literalSources)
 			m_compiler->useMetadataLiteralSources(true);
-		m_compiler->setMetadataHash(m_options.metadata.hash);
-		if (m_options.modelChecker.initialize)
-			m_compiler->setModelCheckerSettings(m_options.modelChecker.settings);
+
+		m_compiler->setMetadataHash(m_options.metadata.hash.value_or(CompilerStack::MetadataHash::IPFS));
```

Comment:
Let's just not call `setMetadataHash()` if the option has no value. Same for `setRevertStringBehaviour()` below.